controllers/subnamespace_controller.go

```diff
@@ -34,18 +36,105 @@ type SubnamespaceReconciler struct {
 	Scheme *runtime.Scheme
 }
 
-// +kubebuilder:rbac:groups=dana.github.com/Dana-Team/SNS,resources=subnamespaces,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=dana.github.com/Dana-Team/SNS,resources=subnamespaces/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=dana.dana.hns.io,resources=subnamespaces,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=dana.dana.hns.io,resources=subnamespaces/status,verbs=get;update;patch
+
+func (r *SubnamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("subspace", req.NamespacedName)
+	var subspace danav1alpha1.Subnamespace
+	var ownerNamespace v1.Namespace
+	var childNamespace v1.Namespace
+	var childNamespaceName string
+
+	// Getting reconciled subspace
+	if err := r.Get(ctx, req.NamespacedName, &subspace); err != nil {
+		log.Error(err, "Could not find Subspace")
+		return ctrl.Result{}, err
+	}
+	// Getting subspace's namespace - described as ownerNamespace
+	if err := r.Get(ctx, client.ObjectKey{Namespace: "", Name: subspace.GetNamespace()}, &ownerNamespace); err != nil {
+		log.Error(err, "Could not find owner Namespace ")
+		return ctrl.Result{}, err
+	}
+
+	// Hierarchy 'breadcrumb' name, built from ownerNamespace and subspace.
+	childNamespaceName = ownerNamespace.Name + "-" + subspace.Name
 
-func (r *SubnamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("subnamespace", req.NamespacedName)
+	// Subspace phase None flow
+	if subspace.Status.Phase == danav1alpha1.None {
 
-	// your logic here
+		if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &subspace, func() error {
+			err := InitializeSubspace(ownerNamespace, subspace, childNamespaceName)
+			return err
+		}); err != nil {
+			log.Error(err, "Could not update Subspace Phase from 'None' to 'Missing'")
+		}
+	}
 
```

Comment:
CreateOrUpdate code is pretty long, do we really want it inside the reconcile function?

Reply:
All of my CreateOrUpdate are 2 lines long. 
Why do you think this one is long ?

Reply:
You update the SNS status two times, from `None` to `Missing` and from `Missing` to `Created`. 
I think you should create a method for this operation.